### ISSUE TITLE
[Bug]: Link as button behaves like a block

### ISSUE THREAD

**Issue by layershifter** [microsoft/fluentui#29600]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/loving-carlos-rzt567?file=/example.tsx

## Actual Behavior

`<Link as="a">` & `<Link as="button">` behave differently:

<img width="465" alt="image" src="https://github.com/microsoft/fluentui/assets/14183168/61f5de32-4a58-4702-afa2-b6f02b148932">

## Expected Behavior

`<Link as="a">` & `<Link as="button">` render the same.

### Logs

```shell
N/A
```

### Requested priority

High

### Products/sites affected

Teams

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by layershifter** [comment]
TL;DR you can't display `button` inline, nothing have changed for years: https://www.scottohara.me/blog/2018/10/03/unbutton-buttons.html

----

<img width="457" alt="image" src="https://github.com/microsoft/fluentui/assets/14183168/bf54f83a-42bb-46da-bcc3-d2793df20b33">

I made a PoC that implements `span role="button"` behavior: https://codesandbox.io/s/withered-snowflake-sw5zll?file=/example.tsx

**Comment by behowell** [comment]
It sounds like the request here is to add support for `<Link as="span">`. We can't change what element is rendered for Link by default (as is done in the PoC codesandbox), since that would be a breaking change; it would need to be opt-in using `as="span"`.

**Comment by khmakoto** [comment]
> It sounds like the request here is to add support for `<Link as="span">`. We can't change what element is rendered for Link by default (as is done in the PoC codesandbox), since that would be a breaking change; it would need to be opt-in using `as="span"`.

Spoke with @layershifter offline and we agreed that this is what we should probably do.
